### PR TITLE
Rework message

### PR DESCRIPTION
**Pull request by OznOg**
use boost for network
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39054676%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39054812%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39056660%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39056691%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39057290%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20388e7fb35956537307ac14db39b4a07014dc1ffa%20remote-processor/Message.h%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39054812%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22maybe%20add%20a%20doxygen%20for%20the%20class.%20What%20is%20is%20doing.%22%2C%20%22created_at%22%3A%20%222015-09-09T15%3A09%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/Message.h%3AL31-59%22%7D%2C%20%22Pull%20388e7fb35956537307ac14db39b4a07014dc1ffa%20.travis.yml%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39056660%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22temporary%20workaround%20%28before%20this%20package%20is%20added%20to%20the%20whitelist%29%20suggestion%3A%5Cr%5Cn%5Cr%5Cnin%20the%20%60install%60%20section%3A%5Cr%5Cn%5Cr%5Cn%7E%7E%7E%5Cr%5Cn-%20wget%20%27https%3A//01.org/sites/default/files/asio-1.10.6.tar.gz%27%5Cr%5Cn-%20tar%20xf%20asio-1.10.6.tar.gz%20-C%20%24PREFIX%20--strip-components%3D1%5Cr%5Cn%7E%7E%7E%22%2C%20%22created_at%22%3A%20%222015-09-09T15%3A23%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20.travis.yml%3AL28-35%22%7D%2C%20%22Pull%20388e7fb35956537307ac14db39b4a07014dc1ffa%20parameter/ParameterMgr.cpp%2026%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39057290%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Proposal%3A%20add%20an%20overloaded%20version%20of%20%60CRequestMessage%3A%3AgetArguments%60%3A%5Cr%5Cn%5Cr%5Cn%20%20%20%20const%20std%3A%3Astring%20%26getArguments%28std%3A%3Avector%3A%3Asize_type%20index%29%20const%20%7B%20return%20_arguments%5Bindex%5D%3B%20%7D%22%2C%20%22created_at%22%3A%20%222015-09-09T15%3A27%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgr.cpp%3AL924-939%22%7D%2C%20%22Pull%20388e7fb35956537307ac14db39b4a07014dc1ffa%20CMakeLists.txt%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39056691%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22unrelated%22%2C%20%22created_at%22%3A%20%222015-09-09T15%3A23%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CMakeLists.txt%3AL43-49%22%7D%2C%20%22Pull%20388e7fb35956537307ac14db39b4a07014dc1ffa%20remote-processor/Message.h%2063%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/188%23discussion_r39054676%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22const%22%2C%20%22created_at%22%3A%20%222015-09-09T15%3A08%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/Message.h%3AL31-59%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 388e7fb35956537307ac14db39b4a07014dc1ffa remote-processor/Message.h 63'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/188#discussion_r39054676'>File: remote-processor/Message.h:L31-59</a></b>
- <a href='https://github.com/krocard'></a> const
- [ ] <a href='#crh-comment-Pull 388e7fb35956537307ac14db39b4a07014dc1ffa remote-processor/Message.h 9'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/188#discussion_r39054812'>File: remote-processor/Message.h:L31-59</a></b>
- <a href='https://github.com/krocard'></a> maybe add a doxygen for the class. What is is doing.
- [ ] <a href='#crh-comment-Pull 388e7fb35956537307ac14db39b4a07014dc1ffa .travis.yml 4'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/188#discussion_r39056660'>File: .travis.yml:L28-35</a></b>
- <a href='https://github.com/dawagner'></a> temporary workaround (before this package is added to the whitelist) suggestion:
in the `install` section:
~~~
- wget 'https://01.org/sites/default/files/asio-1.10.6.tar.gz'
- tar xf asio-1.10.6.tar.gz -C $PREFIX --strip-components=1
~~~
- [ ] <a href='#crh-comment-Pull 388e7fb35956537307ac14db39b4a07014dc1ffa CMakeLists.txt 4'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/188#discussion_r39056691'>File: CMakeLists.txt:L43-49</a></b>
- <a href='https://github.com/dawagner'></a> unrelated
- [ ] <a href='#crh-comment-Pull 388e7fb35956537307ac14db39b4a07014dc1ffa parameter/ParameterMgr.cpp 26'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/188#discussion_r39057290'>File: parameter/ParameterMgr.cpp:L924-939</a></b>
- <a href='https://github.com/dawagner'></a> Proposal: add an overloaded version of `CRequestMessage::getArguments`:
const std::string &getArguments(std::vector::size_type index) const { return _arguments[index]; }

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/188?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/188?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/188'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/188'>Refresh</a>.<a href='#crh-update-none'></a>

File: remote-processor/Message.h [hunk 1]
```diff
@@ -31,103 +31,28 @@
...
-    * @param[in] uiSize array size in bytes
-    */
-    void readData(void* pvData, size_t uiSize);
-
-    /** Write string to the message
-    *
-    * @param[in] strData the string to write
-    */
-    void writeString(const std::string& strData);
+    Result send(const std::vector<uint8_t> &data);
```
Comment (krocard) [inline remote-processor/Message.h]: const

File: remote-processor/Message.h [hunk 2]
```diff
@@ -31,103 +31,28 @@
 
 #include <stdint.h>
 #include <string>
-
-class CSocket;
+#include <vector>
+#include <asio.hpp>
 
 class CMessage
```
Comment (krocard) [inline remote-processor/Message.h]: maybe add a doxygen for the class. What is is doing.

File: .travis.yml [hunk 3]
```diff
@@ -28,6 +28,7 @@ addons:
             - g++-4.8
             - cmake
             - python3-dev
+            - libasio-dev
```
Comment (dawagner) [inline .travis.yml]: temporary workaround (before this package is added to the whitelist) suggestion:

in the `install` section:

~~~
- wget 'https://01.org/sites/default/files/asio-1.10.6.tar.gz'
- tar xf asio-1.10.6.tar.gz -C $PREFIX --strip-components=1
~~~

File: CMakeLists.txt [hunk 4]
```diff
@@ -43,7 +43,6 @@ option(REQUIREMENTS "Generate the html version of the 'requirements' documentati
 option(PYTHON_BINDINGS "Python library to use the Parameter Framrwork from python" ON)
 option(C_BINDINGS "Library to use the Parameter Framework using a C API" ON)
 
-
```
Comment (dawagner) [inline CMakeLists.txt]: unrelated

File: parameter/ParameterMgr.cpp [hunk 5]
```diff
@@ -924,15 +924,15 @@ CParameterMgr::CCommandHandler::CommandStatus CParameterMgr::statusCommandProces
 }
 
 /// Tuning Mode
-CParameterMgr::CCommandHandler::CommandStatus CParameterMgr::setTuningModeCommandProcess(const IRemoteCommand& remoteCommand, string& strResult)
+CParameterMgr::CCommandHandler::CommandStatus CParameterMgr::setTuningModeCommandProcess(const CRequestMessage& remoteCommand, string& strResult)
 {
-    if (remoteCommand.getArgument(0) == "on") {
+    if (remoteCommand.getArguments()[0] == "on") {
```
Comment (dawagner) [inline parameter/ParameterMgr.cpp]: Proposal: add an overloaded version of `CRequestMessage::getArguments`:

    const std::string &getArguments(std::vector::size_type index) const { return _arguments[index]; }
